Framework/DataHandling/test/LoadSESANSTest.h

```diff
@@ -0,0 +1,242 @@
+    TS_ASSERT_EQUALS(ws->getNumberHistograms(), 1);
+    // One line should have been dropped, as it did not have enough columns
+    TS_ASSERT_EQUALS(ws->getNPoints(), 5);
+
+    double tolerance = 1e-5;
+    // Test the first two rows we read
+    // These values are all hard-coded in the sample file in writeGoodFile(),
+    // using:
+    // Y = e ^ (depol * wavelength ^2)
+    // E = depolError * Y * wavelength ^ 2
+    // X = wavelength
+    TS_ASSERT_DELTA(ws->x(0)[0], 1.612452, tolerance);
+    TS_ASSERT_DELTA(ws->y(0)[0], exp(-1.42e-3 * 1.612452 * 1.612452),
+                    tolerance);
+    TS_ASSERT_DELTA(ws->e(0)[0], 2.04e-3 * ws->y(0)[0] * 1.612452 * 1.612452,
+                    tolerance);
+
+    TS_ASSERT_DELTA(ws->x(0)[1], 1.675709, tolerance);
+    TS_ASSERT_DELTA(ws->y(0)[1], exp(-1.45e-3 * 1.675709 * 1.675709),
+                    tolerance);
+    TS_ASSERT_DELTA(ws->e(0)[1], 1.87e-3 * ws->y(0)[0] * 1.675709 * 1.675709,
+                    tolerance);
+  }
+
+  void test_confidence() {
+    LoadSESANS testAlg;
+    TS_ASSERT_THROWS_NOTHING(testAlg.initialize());
+    TS_ASSERT(testAlg.isInitialized());
+    testAlg.setChild(true);
+    testAlg.setRethrows(true);
+    TS_ASSERT_THROWS_NOTHING(testAlg.setProperty("Filename", infileName));
+    Mantid::Kernel::FileDescriptor descriptor(
+        testAlg.getPropertyValue("Filename"));
+    TS_ASSERT_EQUALS(testAlg.confidence(descriptor), 70);
+  }
+
+  void test_requireFFV() {
+    LoadSESANS testAlg;
+    TS_ASSERT_THROWS_NOTHING(testAlg.initialize());
+    TS_ASSERT(testAlg.isInitialized());
+    testAlg.setChild(true);
+    testAlg.setRethrows(true);
+    writeFileMissingFFV();
+    TS_ASSERT_THROWS_NOTHING(testAlg.setProperty("Filename", infileName));
+    TS_ASSERT_THROWS_NOTHING(testAlg.setProperty("OutputWorkspace", "ws"));
+
+    TS_ASSERT_THROWS(testAlg.execute(), std::runtime_error);
+    deleteFile(testAlg);
+  }
+
+  void test_mandatoryHeaders() {
+    LoadSESANS testAlg;
+    TS_ASSERT_THROWS_NOTHING(testAlg.initialize());
+    TS_ASSERT(testAlg.isInitialized());
+    testAlg.setChild(true);
+    testAlg.setRethrows(true);
+    writeFileMissingHeaders();
+    TS_ASSERT_THROWS_NOTHING(testAlg.setProperty("Filename", infileName));
+    TS_ASSERT_THROWS_NOTHING(testAlg.setProperty("OutputWorkspace", "ws"));
+
+    TS_ASSERT_THROWS(testAlg.execute(), std::runtime_error);
+    deleteFile(testAlg);
+  }
+
+  void test_mandatoryColumns() {
+    LoadSESANS testAlg;
+    TS_ASSERT_THROWS_NOTHING(testAlg.initialize());
+    TS_ASSERT(testAlg.isInitialized());
+    testAlg.setChild(true);
+    testAlg.setRethrows(true);
+    writeFileMissingColumns();
+    TS_ASSERT_THROWS_NOTHING(testAlg.setProperty("Filename", infileName));
+    TS_ASSERT_THROWS_NOTHING(testAlg.setProperty("OutputWorkspace", "ws"));
+
+    TS_ASSERT_THROWS(testAlg.execute(), std::runtime_error);
+    deleteFile(testAlg);
+  }
+
+private:
+  std::string infileName = "temp.ses";
```

Comment:
Would [this](https://pocoproject.org/docs/Poco.TemporaryFile.html) be useful here? I only recently learned about this, but it seems to be a good approach, especially when several parallel might create/delete temporary files with the same name. This has led in the past to spurious build server failures